frontend/src/metabase/visualizations/visualizations/Table.jsx

```diff
@@ -130,10 +130,14 @@ export default class Table extends Component {
             data: { cols, rows },
           },
         ],
-        settings,
+        { "table.pivot": pivotCol },
       ) => {
-        const col = cols.filter(isMetric)[0];
-        return col && col.name;
+        // We try to show numeric values in pivot cells, but if none are
+        // available, we fall back to the last column in the unpivoted table
+        const nonPivotCols = cols.filter(c => c.name !== pivotCol);
+        const lastCol = nonPivotCols[nonPivotCols.length - 1];
```

Comment:
I was iffy on my choice to default to the last column if there are no metrics available. My thinking was that it seems more common to have grouped columns as the first two.